test/plugins/templates.test.js

```diff
@@ -432,6 +434,46 @@ describe('template plugin test', () => {
         });
     });
 
+    describe('GET /templates/name/versionOrTag/usage/pipelines', () => {
+        let options;
+
+        beforeEach(() => {
+            options = {
+                method: 'GET',
+                url: '/templates/screwdriver%2Fbuild/1.7.3/usage/pipelines'
+            };
+        });
+
+        it('returns 200 and all pipelines that use the template version', () => {
+            templateFactoryMock.getPipelineUsage.resolves(testPipelineUsage);
+
+            return server.inject(options).then(reply => {
+                assert.deepEqual(reply.result, testPipelineUsage);
+                assert.equal(reply.statusCode, 200);
+                assert.calledWith(templateFactoryMock.getPipelineUsage, `screwdriver/build@1.7.3`);
+            });
+        });
+
+        it('returns 200 even if no pipelines are using the template version', () => {
+            templateFactoryMock.getPipelineUsage.resolves([]);
+
+            return server.inject(options).then(reply => {
+                assert.deepEqual(reply.result, []);
+                assert.equal(reply.statusCode, 200);
+                assert.calledWith(templateFactoryMock.getPipelineUsage, `screwdriver/build@1.7.3`);
+            });
+        });
+
+        it('returns 404 when the template does not exist', () => {
+            templateFactoryMock.getPipelineUsage.resolves(null);
```

Comment:
This would change.
https://github.com/screwdriver-cd/models/pull/583/files#r1312062364